src/status_im2/contexts/wallet/events.cljs

```diff
@@ -4,34 +4,103 @@
+              :fx [[:dispatch [:navigate-to :wallet-accounts address]]]}
+
+       new-account?
+       (update :fx conj [:dispatch [:wallet/show-account-created-toast address]])))))
+
+(rf/reg-event-fx :wallet/close-account-page
+ (fn [{:keys [db]}]
+   {:db (update db :wallet dissoc :current-viewing-account-address)
+    :fx [[:dispatch [:navigate-back]]]}))
+
+(rf/reg-event-fx :wallet/get-accounts-success
+ (fn [{:keys [db]} [accounts]]
+   (let [wallet-db           (get db :wallet)
+         new-account?        (:new-account? wallet-db)
+         navigate-to-account (:navigate-to-account wallet-db)]
+     (cond-> {:db (reduce (fn [db {:keys [address] :as account}]
+                            (assoc-in db [:wallet :accounts address] account))
+                          db
+                          (data-store/rpc->accounts accounts))
+              :fx [[:dispatch [:wallet/get-wallet-token]]]}
+
+       new-account?
+       (update :fx
+               conj
+               [:dispatch [:hide-bottom-sheet]]
+               [:dispatch-later
+                [{:dispatch [:navigate-back]
+                  :ms       100}
+                 {:dispatch [:wallet/navigate-to-account navigate-to-account]
+                  :ms       300}]])))))
+
+(rf/reg-event-fx :wallet/get-accounts
+ (fn [_]
+   {:fx [[:json-rpc/call
+          [{:method     "accounts_getAccounts"
+            :on-success [:wallet/get-accounts-success]
+            :on-error   #(log/info "failed to get accounts "
+                                   {:error %
+                                    :event :wallet/get-accounts})}]]]}))
+
+(rf/reg-event-fx
+ :wallet/save-account
+ (fn [_ [{:keys [account callback]}]]
+   {:fx [[:json-rpc/call
+          [{:method     "accounts_saveAccount"
+            :params     [(data-store/<-account account)]
+            :on-success (fn []
+                          (rf/dispatch [:wallet/get-accounts])
+                          (when (fn? callback)
+                            (callback)))
+            :on-error   #(log/info "failed to save account "
+                                   {:error %
+                                    :event :wallet/save-account})}]]]}))
+
+(rf/reg-event-fx :wallet/get-wallet-token
+ (fn [{:keys [db]}]
+   (let [addresses (map :address (vals (get-in db [:wallet :accounts])))]
+     {:fx [[:json-rpc/call
+            [{:method     "wallet_getWalletToken"
+              :params     [addresses]
+              :on-success [:wallet/get-wallet-token-success]
+              :on-error   #(log/info "failed to get wallet token "
+                                     {:error  %
+                                      :event  :wallet/get-wallet-token
+                                      :params addresses})}]]]})))
+
+(rf/reg-event-fx :wallet/get-wallet-token-success
+ (fn [{:keys [db]} [tokens]]
+   {:db (assoc db
+               :wallet/tokens          tokens
+               :wallet/tokens-loading? false)}))
```

Comment:
We are using `:wallet/tokens` instead of a route `[:wallet :tokens]` I guess this is because there are other uses of the namespaced keywords for the wallet namespaces. 
Is it too hard to remove them?

Reply:
Yes, we need to refactor the token data. I have created a separate issue https://github.com/status-im/status-mobile/issues/17823 to handle it.